zerver/lib/webhooks/common.py

```diff
@@ -280,3 +284,34 @@ def parse_multipart_string(body: str) -> dict[str, str]:
         data[field_name] = body
 
     return data
+
+
+def validate_webhook_signature(
+    request: HttpRequest, payload: str, signature: str, algorithm: str = "sha256"
+) -> None:
+    if not settings.VERIFY_WEBHOOK_SIGNATURES:  # nocoverage
+        return
+
+    if algorithm not in hashlib.algorithms_available:
+        raise AssertionError(
+            _("The algorithm '{algorithm}' is not supported.").format(algorithm=algorithm)
+        )
+
+    webhook_secret: str | None = request.GET.get("webhook_secret")
```

Comment:
As Alex [pointed out](https://chat.zulip.org/#narrow/channel/127-integrations/topic/Stripe.3A.20Verify.20webhook.20signatures.20in.20incoming.20payloads/near/2151361), getting `webhook_secret` from the request doesn’t verify anything meaningful; an attacker sending a forged payload can also send their own forged `webhook_secret` value.